tidy3d/components/geometry/polyslab.py

```diff
@@ -137,7 +140,7 @@ def no_complex_self_intersecting_polygon_at_reference_plane(cls, val, values):
         For 2), we heal the polygon, and warn that the polygon has been cleaned up.
         """
         # no need to validate anything here
-        if isclose(values["dilation"], 0):
+        if np.isclose(values["dilation"], 0):
```

Comment:
So at some point we purposefully changed some validators to use `isclose` instead of `np.isclose` because the latter creates a numpy array which can have significant overhead if you e.g. have a metalens with a million polyslabs. We have [this test](https://github.com/flexcompute/tidy3d-core/blob/000aab3e18c9648fcee4ef93195bb76aaaed5a48/tests/test_simulation.py#L8-L38) on the backend about this (it could easily be moved to the frontend if we wanted). I tried a few times using `pre/2.8` and it would run in about 6.5-7s. With your branch, it runs in about 8.5-9s, which is potentially acceptable but still somewhat significant. Increasing the number of geometries seems to have a linear scaling so it's always about 30% slower.

This consideration was definitely much more important in some of the mesher code, where the scaling is with number of structures squared. But yeah even here it has an effect.

Is it necessary to change to numpy?

Reply:
I changed them all back and things are passing, so fine by me. I think originally I had changed them because I was running into validation errors with `slab_bounds` and it was really hard to pinpoint what was failing. and so I ended up basically changing a ton of things and removing validators before I found it. this must have slipped into the PR.